custom/crediblex/portfolio/account/src/main/java/com/crediblex/fineract/portfolio/account/service/CustomStandingInstructionReadPlatformServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+                                                            DatabaseSpecificSQLGenerator sqlGenerator, PaginationHelper paginationHelper) {
+        super(jdbcTemplate, clientReadPlatformService, officeReadPlatformService, portfolioAccountReadPlatformService,
+                dropdownReadPlatformService, columnValidator, sqlGenerator, paginationHelper);
+        this.jdbcTemplate = jdbcTemplate;
+        this.sqlGenerator = sqlGenerator;
+    }
+
+    @Override
+    public StandingInstructionDuesData retriveLoanDuesData(final Long loanId) {
+        final StandingInstructionLoanDuesMapper rm = new StandingInstructionLoanDuesMapper();
+        final String sql = "select " + rm.schema() + " where ml.id= ? and ls.duedate <= " + sqlGenerator.currentBusinessDate()
+                + " and ls.completed_derived = false";
+        return this.jdbcTemplate.queryForObject(sql, rm, new Object[] { loanId }); // NOSONAR
+    }
+
+    private static final class StandingInstructionLoanDuesMapper implements RowMapper<StandingInstructionDuesData> {
+
+        private final String schemaSql;
+
+        StandingInstructionLoanDuesMapper() {
+            final StringBuilder sqlBuilder = new StringBuilder(400);
+
+            sqlBuilder.append("max(ls.duedate) as dueDate,sum(ls.principal_amount) as principalAmount,");
+            sqlBuilder.append("sum(ls.principal_completed_derived) as principalCompleted,");
+            sqlBuilder.append("sum(ls.principal_writtenoff_derived) as principalWrittenOff,");
+            sqlBuilder.append("sum(ls.interest_amount) as interestAmount,");
+            sqlBuilder.append("sum(ls.interest_completed_derived) as interestCompleted,");
+            sqlBuilder.append("sum(ls.interest_writtenoff_derived) as interestWrittenOff,");
+            sqlBuilder.append("sum(ls.interest_waived_derived) as interestWaived,");
+            sqlBuilder.append("sum(ls.penalty_charges_amount) as penalityAmount,");
+            sqlBuilder.append("sum(ls.penalty_charges_completed_derived) as penalityCompleted,");
+            sqlBuilder.append("sum(ls.penalty_charges_writtenoff_derived)as penaltyWrittenOff,");
+            sqlBuilder.append("sum(ls.penalty_charges_waived_derived) as penaltyWaived,");
+            sqlBuilder.append("sum(ls.fee_charges_amount) as feeAmount,");
+            sqlBuilder.append("sum(ls.fee_charges_completed_derived) as feecompleted,");
+            sqlBuilder.append("sum(ls.fee_charges_writtenoff_derived) as feeWrittenOff,");
+            sqlBuilder.append("sum(ls.fee_charges_waived_derived) as feeWaived ");
+            sqlBuilder.append("from m_loan_repayment_schedule ls ");
+            sqlBuilder.append(" join m_loan ml on ml.id = ls.loan_id ");
+
+            this.schemaSql = sqlBuilder.toString();
+        }
+
+        public String schema() {
+            return this.schemaSql;
+        }
+
```

Comment:
You need to run spotless and bug checks add the apache licences at to the file. Probably best to set your IDE to do auto add this. @hassanmehdi98 you need to modify the pipeline to run these checks